oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -92,7 +107,8 @@ private UserAuthorizer(
       HttpTransportFactory transportFactory,
       URI tokenServerUri,
       URI userAuthUri,
-      PKCEProvider pkce) {
+      PKCEProvider pkce,
+      ClientAuthenticationType clientAuthentication) {
```

Comment:
Since this is a private constructor, can we update this to take in the Builder class instead of additional parameters? 

```
private UserAuthorizer(Builder builder) {
  // Assign the fields.
}
```

Reply:
I believe you can change the builder's build() method so that it pass the builder as the sole parameter to the constructor:
```
    public UserAuthorizer build() {
      return new UserAuthorizer(this);
    }
```

I think this would be cleaner because future implementations can add additional params without having to explicitly new fields to the constructor:
```
  private UserAuthorizer(UserAuthorizer.Builder builder) {
    this.clientId = Preconditions.checkNotNull(builder.clientId);
    this.scopes = ImmutableList.copyOf(Preconditions.checkNotNull(builder.scopes));
    this.callbackUri = (builder.callbackUri == null) ? DEFAULT_CALLBACK_URI : builder.callbackUri;
    this.transportFactory =
        (builder.transportFactory == null) ? OAuth2Utils.HTTP_TRANSPORT_FACTORY : builder.transportFactory;
    this.tokenServerUri = (builder.tokenServerUri == null) ? OAuth2Utils.TOKEN_SERVER_URI : builder.tokenServerUri;
    this.userAuthUri = (builder.userAuthUri == null) ? OAuth2Utils.USER_AUTH_URI : builder.userAuthUri;
    this.tokenStore = (builder.tokenStore == null) ? new MemoryTokensStorage() : builder.tokenStore;
    this.pkce = builder.pkce;
    this.clientAuthenticationType =
        (builder.clientAuthenticationType == null)
            ? ClientAuthenticationType.CLIENT_SECRET_POST
            : builder.clientAuthenticationType;
  }
```

I think this can be refactored even more, so that the builder sets default values and it doesn't need to be validated in the constructor:
```
  public static class Builder {

    private ClientId clientId;
    private TokenStore tokenStore = new MemoryTokensStorage();
    private URI callbackUri = DEFAULT_CALLBACK_URI;
    private URI tokenServerUri = OAuth2Utils.TOKEN_SERVER_URI;
    private URI userAuthUri = OAuth2Utils.USER_AUTH_URI;
    private Collection<String> scopes;
    private HttpTransportFactory transportFactory = OAuth2Utils.HTTP_TRANSPORT_FACTORY;
    private PKCEProvider pkce;
    private ClientAuthenticationType clientAuthenticationType = ClientAuthenticationType.CLIENT_SECRET_POST;
...
```

and the constructor could be something like:
```
  private UserAuthorizer(UserAuthorizer.Builder builder) {
    this.clientId = Preconditions.checkNotNull(builder.clientId);
    this.scopes = ImmutableList.copyOf(Preconditions.checkNotNull(builder.scopes));
    this.callbackUri = builder.callbackUri;
    this.transportFactory = builder.transportFactory;
    this.tokenServerUri = builder.tokenServerUri;
    this.userAuthUri = builder.userAuthUri;
    this.tokenStore = builder.tokenStore;
    this.pkce = builder.pkce;
    this.clientAuthenticationType = builder.clientAuthenticationType;
  }
```

Reply:
The refactor make sense to me. But that's sort of a bigger refactor than this change and irrelevant to the purpose of this change.
Can I do a separate one for that? Maybe create an issue and resolve it through the refactor change.

Reply:
Sure, it's a private constructor and we can change implementation later. If you could create an issue in this repo and follow up on it sometime in the coming weeks, that would be great! Thanks!